tests/integration_tests/lightgbm_tuner_tests/test_optimize.py

```diff
@@ -732,6 +736,47 @@ def test_optuna_callback(self) -> None:
 
         assert callback_mock.call_count == 10
 
+    def test_tune_best_score_reproducibility(self) -> None:
+        boston = sklearn.datasets.load_boston()
+        X_trainval, X_test, y_trainval, y_test = train_test_split(
+            boston.data, boston.target, random_state=0
+        )
+
+        train = lgb.Dataset(X_trainval, y_trainval)
+        valid = lgb.Dataset(X_test, y_test)
+        params = {"objective": "regression", "metric": "rmse", "random_seed": 0}
+
+        test_study_first_try = optuna.create_study(
+            direction="minimize", sampler=TPESampler(seed=10)
+        )
+        tuner_first_try = lgb.LightGBMTuner(
+            params,
+            train,
+            valid_sets=valid,
+            early_stopping_rounds=3,
+            study=test_study_first_try,
+            optuna_seed=10,
+        )
+        tuner_first_try.run()
+        best_score_first_try = tuner_first_try.best_score
```

Comment:
I think `test_study_first_try` can be removed since all samplers are created with `optuna_seed` inside `LightGBMTuner`. 
```suggestion
        tuner_first_try = lgb.LightGBMTuner(
            params,
            train,
            valid_sets=valid,
            early_stopping_rounds=3,
            optuna_seed=10,
        )
        tuner_first_try.run()
        best_score_first_try = tuner_first_try.best_score
```

Reply:
done